twenty-first/src/util_types/algebraic_prng.rs

```diff
@@ -0,0 +1,37 @@
+use itertools::Itertools;
+
+use crate::shared_math::x_field_element::XFieldElement;
+
+use super::algebraic_hasher::SpongeHasher;
+
+pub trait AlgebraicPrng {
+    type PrngState;
+    fn sample_xfield(state: &mut Self::PrngState, num_elements: usize) -> Vec<XFieldElement>;
+    fn sample_indices(state: &mut Self::PrngState, max: u32, num_elements: usize) -> Vec<u32>;
```

Comment:
Calling the parameter `max: u32` makes it sound like it's an inclusive upper bound, which it is not.

Calling it `upper_bound: u32` leaves it ambiguous, but not indicating towards the wrong interpretation.